### ISSUE TITLE
Recursive cycle of JS user input with certain files

### ISSUE THREAD

**Issue by didroe** [Bearer/bearer-rules#78]
## Description & Reproduction

The `app/assets/javascripts/discourse/app/models/topic.js` file from Discourse is not being scanned due to a cycle error. This means that the same detector is being invoked on the same node in a loop.

The rules need to be changed to avoid this happening.

## Expected Behavior

The file is scanned with no error.

## Actual Behavior

The file is skipped with this log:

```javascript
{"level":"debug","time":"2023-05-25T16:48:39+01:00","message":"failed to process file app/assets/javascripts/discourse/app/models/topic.js for detector: javascript failed to detect in file /Users/dave/projects/bearer/tmp/discourse/app/assets/javascripts/discourse/app/models/topic.js: cycle found in detector usage: [javascript_shared_common_user_input > javascript_shared_common_user_input]\nnode type: shorthand_property_identifier_pattern, content:\ntracked"}
```

## Possible Fix

Change `javascript_shared_common_user_input` to avoid it referencing itself for the same node (other nodes are ok).

## Your Environment

- Operating System and version: N/A
- Output of 'bearer version':
```
bearer version: 1.9.0
sha: 3990de38c7b7b692a438203a08706dc459c3adcf
```
